### PR TITLE
fix: added support in TestRunExistingResourcesInstances  test to use unique prefix

### PR DESCRIPTION
**Pull request by Khuzaima05**
### Description

- added support in TestRunExistingResourcesInstances  test to use unique prefix
- added a validation in cos block to prevent COS creation if existing SCC is passed, to avoid bucket conflict.

issue : https://github.ibm.com/orgs/GoldenEye/projects/3/views/8?filterQuery=assignee%3A%40me&pane=issue&itemId=421724

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by Khuzaima05** [comment]
@ocofaigh , I've resolved all the review comments, could you please review again?

**Comment by ocofaigh** [comment]
@Khuzaima05 The pipeline is failing - please check

File: main.tf [hunk 1]
```diff
@@ -49,7 +49,7 @@ locals {
 }
 
 resource "ibm_iam_authorization_policy" "scc_cos_s2s_access" {
-  count                       = var.skip_cos_iam_authorization_policy ? 0 : 1
+  count                       = var.existing_scc_instance_crn != null || var.skip_cos_iam_authorization_policy ? 0 : 1
```
Comment (ocofaigh) [inline main.tf:52]: I think you should update the variable description of `skip_cos_iam_authorization_policy` to explicitly call out that the value is ignored if passing an existing SCC instance.

**Comment by Khuzaima05** [comment]
@ocofaigh ,
Upgrade test is failing, I have tested it manually apply and re-apply.
This is the change which is happening:
Screenshot 2024-09-30 at 1 44 35 PM

Can I skip the upgrade test?

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@Khuzaima05 In the complete example, you added a count to the cos module. So hence the statefile in the upgrade test now thinks it needs to recreate the COS instance and bucket. Because they are getting recreated, the SCC settings and the auth policy have been identified for updates with the new COS details.

We have 2 options..
1. add a moved block to the example to prevent COS from being recreated
2. skip upgrade test

Usually I would say there is no need to add a moved block to an example, but I think I would like to see the upgrade test pass here to ensure we don’t have any other issues. So suggest to add that to the example (add it in a `moved.tf` file) just for this PR, and we can remove it in a follow up PR.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**